tasks/all-in-one/parameter-sanity-check-task.yaml

```diff
@@ -0,0 +1,124 @@
+          # skipped and the continue with the rest of the Tasks so for now this pipeline
+          # only work if cluster is provisioned as part of its run
+          if [[ "$(params.provision-cluster)" != "true" ]]; then
+            echo "You need to provision a new cluster as part of PipelineRun due to https://github.com/tektoncd/pipeline/issues/7680"
+            exit 1
+          fi
+          # End of workaround
+
+          if [[ "$(params.provision-cluster)" == "true" ]]; then
+            # make sure that there is no cluster info if a new cluster is to be provisioned
+            if [[ "x$(params.kube-api)" != "x" ]]; then
+              echo "'cluster-provision' parameter is set to 'true', the 'kube-api' parameter must be left blank"
+              exit 1
+            fi
+            if [[ "x$(params.cluster-credentials)" != "x" ]]; then
+              echo "'cluster-provision' parameter is set to 'true', the 'cluster-credentials' parameter must be left blank"
+              exit 1
+            fi
+            # make sure proper defaults are used if provider credentials are not specified
+            if [[ "x$(params.provider-credentials)" == "x" ]]; then
+              if [[ "$(params.cluster-type)" == "osd-aws" ]]; then
+                echo -n 'kua-aws-credentials' | tee $(results.provider-credentials.path) && echo ""
+              fi
+              if [[ "$(params.cluster-type)" == "osd-gcp" ]]; then
+                echo -n 'kua-gcp-credentials' | tee $(results.provider-credentials.path) && echo ""
+              fi
+              if [[ "$(params.cluster-type)" == "rosa" ]]; then
+                echo -n 'kua-rosa-credentials' | tee $(results.provider-credentials.path) && echo ""
+              fi
+            else
+              echo -n "$(params.provider-credentials)" | tee $(results.provider-credentials.path) && echo ""
+            fi
+          else
+            # check that cluster info is provided if not provisioning the cluster
+            if [[ "x$(params.kube-api)" == "x" ]]; then
+              echo "'cluster-provision' parameter is not set to 'true', the 'kube-api' parameter must not be left blank"
+              exit 1
+            fi
+            if [[ "x$(params.cluster-credentials)" == "x" ]]; then
+              echo "'cluster-provision' parameter is not set to 'true', the 'cluster-credentials' parameter must not be left blank"
+              exit 1
+            fi
+          fi
+
+          # make sure default flags are used for cluster creation command if these are not specified
+          if [[ "x$(params.create-cmd-flags)" == "x" ]]; then
+            if [[ "$(params.cluster-type)" == "osd-aws" ]]; then
+              cmd_flags='--provider=aws --ccs --channel-group=stable --compute-machine-type=m5.xlarge --compute-nodes=3 --enable-autoscaling=false --multi-az=false --region=us-east-1 --version=4.18.10 --machine-cidr=10.11.128.0/23 --pod-cidr=10.11.64.0/18 --service-cidr=10.11.0.0/18 --host-prefix=23'
```

Comment:
No hardcoded openshift pls "4.18.10"

Reply:
Done, I made OCM login very first Task so in parameter check Task it is possible to get current default version which is used if `--version` flag is not specified.